crates/oxc_language_server/src/commands.rs

```diff
@@ -0,0 +1,115 @@
+};
+
+use crate::{capabilities::Capabilities, Backend};
+
+pub const LSP_COMMANDS: [WorkspaceCommands; 1] = [WorkspaceCommands::FixAll(FixAllCommand)];
+
+pub trait WorkspaceCommand {
+    fn command_id(&self) -> String;
+    fn available(&self, cap: Capabilities) -> bool;
+    type CommandArgs<'a>: serde::Deserialize<'a>;
+    async fn execute<'a>(
+        &self,
+        backend: &Backend,
+        args: Self::CommandArgs<'a>,
+    ) -> jsonrpc::Result<Option<serde_json::Value>>;
+}
+
+pub enum WorkspaceCommands {
+    FixAll(FixAllCommand),
+}
+
+impl WorkspaceCommands {
+    pub fn command_id(&self) -> String {
+        match self {
+            WorkspaceCommands::FixAll(c) => c.command_id().into(),
+        }
+    }
+    pub fn available(&self, cap: Capabilities) -> bool {
+        match self {
+            WorkspaceCommands::FixAll(c) => c.available(cap),
+        }
+    }
+    pub async fn execute(
+        &self,
+        backend: &Backend,
+        args: Vec<serde_json::Value>,
+    ) -> jsonrpc::Result<Option<serde_json::Value>> {
+        match self {
+            WorkspaceCommands::FixAll(c) => {
+                let arg: Result<
+                    <FixAllCommand as WorkspaceCommand>::CommandArgs<'_>,
+                    serde_json::Error,
+                > = serde_json::from_value(serde_json::Value::Array(args));
+                if let Err(e) = arg {
+                    error!("Invalid args passed to {:?}: {e}", c.command_id());
+                    return Err(Error::invalid_request());
+                }
+                let arg = arg.unwrap();
+
+                c.execute(backend, arg).await
+            }
+        }
+    }
+}
+
+pub struct FixAllCommand;
+
+#[derive(Deserialize)]
+pub struct FixAllCommandArg {
+    uri: String,
+}
+
+impl WorkspaceCommand for FixAllCommand {
+    fn command_id(&self) -> String {
+        "oxc.fixAll".into()
+    }
+    fn available(&self, cap: Capabilities) -> bool {
+        cap.workspace_edit
+    }
+    type CommandArgs<'a> = (FixAllCommandArg,);
+
+    async fn execute<'a>(
+        &self,
+        backend: &Backend,
+        args: Self::CommandArgs<'a>,
+    ) -> jsonrpc::Result<Option<serde_json::Value>> {
+        let url = Url::parse(&args.0.uri);
+        if let Err(e) = url {
+            error!("Invalid uri passed to {:?}: {e}", self.command_id());
+            return Err(Error::invalid_request());
+        }
+        let url = url.unwrap();
+
+        let mut edits = vec![];
+        if let Some(value) = backend.diagnostics_report_map.get(&url.to_string()) {
+            for report in value.iter() {
+                if let Some(fixed) = &report.fixed_content {
+                    edits.push(TextEdit { range: fixed.range, new_text: fixed.code.clone() })
+                }
+            }
+            let _ = backend
+                .client
+                .send_request::<ApplyWorkspaceEdit>(ApplyWorkspaceEditParams {
+                    label: Some("Oxlint fix all".into()),
```

Comment:
This message is shown to the user (nvim) and in the undo history (vscode). Maybe better label is applicable (such as Oxlint: {n} fixes applied)

Reply:
I could not tests this with vscode. Could to provide me the shortcude `CTRL + Shift + P` did not work.
I guess maybe we need to extend the `package.json` information for vscode like eslint did:

https://github.com/microsoft/vscode-eslint/blob/main/package.json#L564-L595

My test setup:
```
cd editors/vscode
pnpm build
pnpm install-extension
// restart vscode, but wait a bit to clear the cache
```